src/CaptureFile/ProtoSectionInputStreamImpl.cpp

```diff
@@ -12,12 +12,22 @@ namespace orbit_capture_file_internal {
 constexpr uint64_t kMaximumMessageSize = 1024 * 1024;  // 1Mb
 
 ErrorMessageOr<void> ProtoSectionInputStreamImpl::ReadMessage(google::protobuf::Message* message) {
+  // CodedInputStream imposes a hard limit on the total number of bytes it will read. It's INT_MAX
+  // by default and it cannot be increased past that. To work around the limitation, reinitialize
+  // the CodedInputStream, as the actual current position is kept by the FileFragmentInputStream
+  // instead. Note that this makes CodedInputStream::CurrentPosition not always reflect the actual
+  // position in the stream.
+  if (coded_input_stream_->CurrentPosition() >= kCodedInputStreamReinitializationThreshold) {
+    coded_input_stream_.emplace(&file_fragment_input_stream_);
+    coded_input_stream_->SetTotalBytesLimit(kCodedInputStreamTotalBytesLimit);
```

Comment:
out of curiosity, why is this call needed?

Reply:
It's not needed. But since `kCodedInputStreamReinitializationThreshold`'s value depends on the maximum, we want to make clear that we don't rely on previous knowledge that the default maximum is INT_MAX, we just set the maximum ourselves again.